Sources/Commands/SwiftTestCommand.swift

```diff
@@ -1021,33 +1027,27 @@ final class TestRunner {
         return args
     }
 
-    private func test(at path: AbsolutePath, outputHandler: @escaping (String) -> Void) -> Result {
+    private func test(at path: AbsolutePath, outputHandler: @escaping (String) -> Void) async -> Result {
         let testObservabilityScope = self.observabilityScope.makeChildScope(description: "running test at \(path)")
 
         do {
-            let outputHandler = { (bytes: [UInt8]) in
-                if let output = String(bytes: bytes, encoding: .utf8) {
-                    outputHandler(output)
-                }
-            }
-            let outputRedirection = AsyncProcess.OutputRedirection.stream(
-                stdout: outputHandler,
-                stderr: outputHandler
-            )
-            let process = AsyncProcess(arguments: try args(forTestAt: path), environment: self.testEnv, outputRedirection: outputRedirection)
-            guard let terminationKey = self.cancellator.register(process) else {
-                return .failure // terminating
+            let args = try args(forTestAt: path)
+            let processConfig = try Subprocess.Configuration(commandLine: args, environment: .init(self.testEnv))
+            let status = try await cancellator.run(name: "Test Execution") {
+                try await Subprocess.run(processConfig, input: .none, error: .standardOutput) { execution, outputSequence in
```

Comment:
Shouldn't this be `error: .combineWithOutput`? (see swiftlang/swift-subprocess#180)

Reply:
yeah, that's what I was looking for